backend/endpoints/rooms.py

```diff
@@ -41,19 +41,22 @@ async def create_room(
         HTTP_200_OK (room already exist): {room_id}
         HTTP_201_CREATED (new room created): {room_id}
         HTTP_424_FAILED_DEPENDENCY: room creation unsuccessful
+        HTTP_406_NOT_ACCEPTABLE: max number of dm users exceeded
     """
     room_data = request.dict()
-    room_member_ids = room_data["room_members"]
-    user_rooms = await get_rooms(member_id, org_id)
-    print("rooms", user_rooms)
+    room_members = set(room_data["room_members"])
+    if len(room_members) > 9:
+        return JSONResponse(
+            content={"message": "Cannot have more than 9 users in a DM"},
+            status_code=status.HTTP_406_NOT_ACCEPTABLE,
+        )
+    user_rooms = await get_rooms(org_id)  # add member_id after adding it to function
     if isinstance(user_rooms, list):
         for room in user_rooms:
-            room_users = room["room_members"]
-            if dict(room_users).keys == dict(room_member_ids).keys:
+            if room_members == set(room_data["room_members"]):
                 return JSONResponse(
                     content={"room_id": room["_id"]}, status_code=status.HTTP_200_OK
                 )
-
     elif user_rooms is None:
         return JSONResponse(
             content="unable to read database", status=status.HTTP_424_FAILED_DEPENDENCY
```

Comment:
Here we could use HTTPException class to handle errors.

An example here: https://fastapi.tiangolo.com/tutorial/handling-errors/

Reply:
I have been able to use HTTPException where necessary in my recent commit.
Thank you Zxenon for the resources, it was really helpful

Reply:
Great.